spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/messaging/CursorReadingTask.java

```diff
@@ -39,7 +41,7 @@
  */
 abstract class CursorReadingTask<T, R> implements Task {
 
-	private final Object lifecycleMonitor = new Object();
```

Comment:
`synchronized` blocks within this class serve the purpose to ensure ordering according (happens before as of the JLS spec). We should keep these to avoid visibility issues of the `state` field.